src/pages/trade/ui/positions.tsx

```diff
@@ -39,7 +56,132 @@ const ErrorNotice = ({ error }: { error: unknown }) => {
   );
 };
 
-const PositionsInner = observer(() => {
+const getStateLabel = (
+  state: PositionState_PositionStateEnum,
+  side?: Order['side'],
+): { label: string; color: TextProps['color'] } => {
+  if (side && state === PositionState_PositionStateEnum.OPENED) {
+    if (side === 'Buy') {
+      return { label: side, color: 'success.light' };
+    } else {
+      return { label: side, color: 'destructive.light' };
+    }
+  } else {
+    return { label: stateToString(state), color: 'neutral.light' };
+  }
+};
+
+const ActionButton = observer(
+  ({ state, id }: { state: PositionState_PositionStateEnum; id: PositionId }) => {
+    const { loading, closePositions, withdrawPositions } = positionsStore;
+
+    if (state === PositionState_PositionStateEnum.OPENED) {
+      return (
+        <Button onClick={() => void closePositions([id])} disabled={loading}>
+          Close
+        </Button>
+      );
+    } else if (state === PositionState_PositionStateEnum.CLOSED) {
+      return (
+        <Button disabled={loading} onClick={() => void withdrawPositions([id])}>
+          Withdraw
+        </Button>
+      );
+    } else {
+      return (
+        <Text detail color='text.secondary'>
+          -
+        </Text>
+      );
+    }
+  },
+);
+
+const RowLabel = ({ p }: { p: PositionData }) => {
+  // A withdrawn position has no orders
+  if (!p.orders.length) {
+    const { label, color } = getStateLabel(p.positionState);
+    return (
+      <Text detail color={color}>
+        {label}
+      </Text>
+    );
+  }
+  // For opened or closed positions
+  return p.orders.map((o, i) => {
+    const { label, color } = getStateLabel(p.positionState, o.side);
+    return (
+      <Text detail color={color} key={i}>
+        {label}
+      </Text>
+    );
+  });
+};
+
+const AmountDisplay = ({
+  p,
+  kind,
+}: {
+  p: PositionData;
+  kind: 'tradeAmount' | 'effectivePrice';
+}) => {
+  if (!p.orders.length) {
+    return (
+      <Text detail color='text.secondary'>
+        -
+      </Text>
+    );
+  }
+  return p.orders.map((o, i) => (
+    <ValueViewComponent
+      valueView={kind === 'tradeAmount' ? o.tradeAmount : o.effectivePrice}
+      key={i}
+    />
+  ));
+};
+
+const MAX_ACTION_COUNT = 15;
+
+const HeaderActionButton = observer(() => {
+  const { data } = usePositions();
```

Comment:
should the `useQuery` have a [retry](https://tanstack.com/query/latest/docs/framework/react/guides/query-retries)?

Reply:
The default actually retries like 5 times I think. We should figure out a way to change this default to retry less because the loading states are too long in a failure case imo. Going to change this to `retry: 1` for now.